packages/package-json-lint/src/rules/require-no-hard-coded-dependency-versions.js

```diff
@@ -0,0 +1,23 @@
+module.exports = {
+  create: ({ ruleConfig, projectType, report }) => ({
+    dependencies: (dependencies) => {
+      if (projectType === 'module' || projectType === 'devModule') {
+        const messageString = 'no hard coded dependency';
+        const currentProblems = Object.keys(dependencies).map(dependencyName => {
+          const dependencyVersion = dependencies[dependencyName];
+          if (!dependencyVersion.startsWith('^') && !(ruleConfig.severity.allowList && ruleConfig.severity.allowList.includes(dependencyName))) {
```

Comment:
Some project dependencies may look like this that could reference a range, the latest snapshot, or renaming the dependency where `^` is not always the first character. Is the recommendation for teams to just add such dependencies in the allowList?

- `"blood-loss-calculator-js": ">= 2.3.0-dev.0 <2.3.0-dev.x"`
- `"ion-provider-relationship-js": "master"`
- `"procedures-js": "dev"`
- ["real-wicg-inert": "npm:wicg-inert@^3.0.2"](https://github.cerner.com/orion/dynamic-component-catalog-toolkit/blob/1.17.0/package.json#L102)

Reply:
I have added a compatibility regex array, through which I am checking whether the dependency has a compatibility version or not.

Should we add `dev`, `master`, or `latest` to the regex array? As they seem like hardcoded dependencies. 
Any thoughts?